src/ApiBundle/Controller/AuthController.php

```diff
@@ -503,12 +440,30 @@ public function editProfileAction()
 
         $data = $request->request->all();
 
-        $this->handleKeyUsername($user, $request);
-        $this->handleKeyEmail($user, $request);
-        $this->handleKeyFirstname($user, $request);
-        $this->handleKeyLastname($user, $request);
-        $this->handleKeyDob($user, $request);
+        $user->setFirstname(array_key_exists('firstname', $data) ? $data['firstname'] : $user->getFirstname() );
+        $user->setLastname(array_key_exists('lastname', $data) ? $data['lastname'] : $user->getLastname() );
+        $user->setDob( array_key_exists('dob', $data) ? $data['dob'] : $user->getDob() );
+
+        // Validate user data
+        $validator = $this->get('validator');
+        $errors = $validator->validate($user, null, array('profile_edit'));
+
+        if (count($errors) > 0) {
+            $errorArray = [];
+            foreach ($errors as $error) {
+                $constraint = $error->getConstraint();
+                $errorItem = array(
+                                    "code" => 400,
+                                    "error" =>  "Bad Request",
+                                    "error_description" => $error->getMessage(),
+                                    "show_message" => $this->get('translator')->trans($constraint->payload['api_error'], array(), 'messages', $request->getLocale())
+                                  );
+                array_push($errorArray, $errorItem);
+            }
+            return new JsonResponse($errorArray);
+          }
 
+        // Everything ok, now update the user record
         $userManager->updateUser($user);
```

Comment:
move         ` $userManager = $this->get('fos_user.user_manager'); ` line on top of this;